src/hotspot/share/code/relocInfo.hpp

```diff
@@ -980,27 +1034,23 @@ class metadata_Relocation : public DataRelocation {
   // an metadata in the CodeBlob's metadata pool
   static RelocationHolder spec(int metadata_index, int offset = 0) {
     assert(metadata_index > 0, "must be a pool-resident metadata");
-    RelocationHolder rh = newHolder();
-    new(rh) metadata_Relocation(metadata_index, offset);
-    return rh;
+    return RelocationHolder::construct<metadata_Relocation>(metadata_index, offset);
   }
   // an metadata in the instruction stream
   static RelocationHolder spec_for_immediate() {
-    const int metadata_index = 0;
-    const int offset    = 0;    // if you want an offset, use the metadata pool
-    RelocationHolder rh = newHolder();
-    new(rh) metadata_Relocation(metadata_index, offset);
-    return rh;
+    return RelocationHolder::construct<metadata_Relocation>(0, 0);
```

Comment:
I'd rather see (metadata_index, offset) than (0, 0), but I guess the meaning of the argument can be found in spec() above.